### PR TITLE
show completed activities

### PR DESCRIPTION
**Pull request by ecornamu**
Added the possibility to tick "completed activities" in the filter screen to show only activities with known position

### CODE REVIEW

File: app/src/main/java/com/lastaoutdoor/lasta/ui/navigation/MainNavGraph.kt [hunk 1]
```diff
@@ -228,13 +228,15 @@ fun NavGraphBuilder.addMainNavGraph(navController: NavHostController) {
           weatherViewModel::fetchWeatherWithUserLoc)
     }
     composable(DestinationRoute.Filter.route) { entry ->
-      val discoverScreenViewModel: DiscoverScreenViewModel = hiltViewModel(entry)
+      val discoverScreenViewModel: DiscoverScreenViewModel = entry.sharedViewModel(navController)
```
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/navigation/MainNavGraph.kt:231]: This was the main cause of our troubles... well played finding it!

File: app/src/main/java/com/lastaoutdoor/lasta/viewmodel/DiscoverScreenViewModel.kt
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/viewmodel/DiscoverScreenViewModel.kt:1]: Nice way of doing so!

File: app/src/main/java/com/lastaoutdoor/lasta/viewmodel/DiscoverScreenViewModel.kt [hunk 2]
```diff
@@ -263,6 +268,12 @@ constructor(
     _range.value = range
   }
 
+    // toggle wether we show completed activities or not
+  fun setShowCompleted(showCompleted: Boolean) {
+    _showCompleted.value = showCompleted
+    fetchActivities()
```
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/viewmodel/DiscoverScreenViewModel.kt:274]: The `fetchActivities()` here is necessary, thank you for pointing it out :)

**Review by Thimphou — APPROVED** [review]
Thank you for these quick fixes, just make sure to format your code, the CI is actually crashing.

**Review by npfrei — APPROVED** [review]
Nice fix ! It was a problem that we had for a long time, thanks for finally fixing it !

**Pull request merged**